jax/numpy/lax_numpy.py

```diff
@@ -1145,19 +1145,16 @@ def squeeze(a, axis: Union[int, Tuple[int, ...]] = None):
   if axis is None:
     a_shape = shape(a)
     axis = tuple(i for i, d in enumerate(a_shape) if d == 1)
-  elif isinstance(axis, int):
+  elif not isinstance(axis, tuple):
     axis = (axis,)
   return lax.squeeze(a, axis)
 
 
 @_wraps(np.expand_dims)
 def expand_dims(a, axis: Union[int, Tuple[int, ...]]):
-  # TODO(https://github.com/google/jax/issues/3243)
-  try:
-    axis = (int(axis),)  # type: ignore[arg-type]
-  except TypeError:
-    pass
-  return lax.expand_dims(a, axis)  # type: ignore[arg-type]
+  if not isinstance(axis, tuple):
```

Comment:
I am wondering how is this code going to behave if somebody passes a DeviceArray for axis? Specifically, I am wondering if the rollback that I hit on Friday will hit again. 

In general, for changes that may break clients, even abusing ones, we should be aware that the breakage will be manifested through an import failure. It would be best to try such a PR inside google3 (go/jax-source-sync) or at least to alert the oncall.

Reply:
I was lazy here with my tests. I used a scalar NumPy array instead of a DeviceArray because you can’t put DeviceArray objects in decorators. But they work the same in this context (converted via operator.index)